File: StarvingFish/GameViewController.m
//
//  GameViewController.m
//  StarvingFish
//
//  Created by Paker on 04/06/16.
//  Copyright (c) 2016 Paker. All rights reserved.
//

#import "GameViewController.h"
#import "GameScene.h"

@implementation GameViewController

- (void)viewDidLoad
{
    [super viewDidLoad];

    // Configure the view.
    SKView * skView = (SKView *)self.view;
//    skView.showsFPS = YES;
//    skView.showsNodeCount = YES;
    /* Sprite Kit applies additional optimizations to improve rendering performance */
    skView.ignoresSiblingOrder = YES;
    
    // Create and configure the scene.
    GameScene *scene = [GameScene sceneWithSize:skView.frame.size];
    scene.scaleMode = SKSceneScaleModeAspectFit;
    
    // Present the scene.
    [skView presentScene:scene];
}

@end

File: StarvingFish/GameViewController.swift
//
//  GameViewController.swift
//  StarvingFish
//
//  Created by Paker on 15/07/23.
//  Copyright © 2023 Paker. All rights reserved.
//

import UIKit
import SpriteKit

class GameViewController: UIViewController {

    var skView: SKView { view as! SKView }

    override func loadView() {
        view = SKView()

//        skView.showsFPS = true
//        skView.showsNodeCount = true

        // Sprite Kit applies additional optimizations to improve rendering performance
        skView.ignoresSiblingOrder = true
    }

    override func viewDidLayoutSubviews() {

        super.viewDidLayoutSubviews()

        guard skView.scene == nil else { return }

        let scene = GameScene(size: skView.frame.size)

        skView.presentScene(scene)
    }
}
